src/RefractDataStructure.cc

```diff
@@ -204,20 +246,33 @@ namespace drafter {
 
         template <typename U, bool dummy = true>
         struct Fetch {
-            U operator()(const mson::TypeSection& t, const mson::BaseTypeName& defaultNestedType) {
-                return LiteralTo<U>(t.content.value);
+            U operator()(const SectionInfo<mson::TypeSection>& t, const mson::BaseTypeName& defaultNestedType) {
+                return LiteralTo<U>(t.section.content.value);
             }
         };
 
         template<bool dummy>
         struct Fetch<RefractElements, dummy> {
-            RefractElements operator()(const mson::TypeSection& t, const mson::BaseTypeName& defaultNestedType) {
-                return MsonElementsToRefract(t.content.elements(), defaultNestedType);
+            RefractElements operator()(const SectionInfo<mson::TypeSection>& t, const mson::BaseTypeName& defaultNestedType) {
+                return MsonElementsToRefract(MakeSectionInfo(t.section.content.elements(), t.sourceMap.elements(), t.hasSourceMap()), defaultNestedType);
             }
         };
 
         template <typename U, bool dummy = true>
-        struct FetchTypeDefinition {};
+        struct FetchSourceMap {
+            snowcrash::SourceMap<U> operator()(const SectionInfo<mson::TypeSection>& t, const mson::BaseTypeName& defaultNestedType) {
+                // conversion of source map from "string" into "typed" sourcemap
+                if (!t.hasSourceMap()) {
+                    return SectionInfo<U>::NullSourceMap();
+                }
+
+                snowcrash::SourceMap<U> sourceMap;
+                sourceMap.sourceMap = t.sourceMap.value.sourceMap;
+                return sourceMap;
+            }
+        };
+
+        template <typename U, bool dummy = true> struct FetchTypeDefinition;
```

Comment:
Please move `struct FetchTypeDefinition` to a new line and leave the first line just for the template definition.